miniapps/meshing/minimal-surface.cpp

```diff
@@ -0,0 +1,1302 @@
+//               -----------------------
+//
+// This miniapp solves Plateau's problem: the Dirichlet problem for the minimal
+// surface equation.
+//
+// Two problems can be run:
+//
+//  - Problem 0 solves the minimal surface equation of parametric surfaces.
+//              The surface (-s) option allow the selection of different
+//              parameterizations:
+//                s=0: Uses the given mesh from command line options
+//                s=1: Catenoid
+//                s=2: Helicoid
+//                s=3: Enneper
+//                s=4: Hold
+//                s=5: Costa
+//                s=6: Shell
+//                s=7: Scherk
+//                s=8: FullPeach
+//                s=9: QuarterPeach
+//               s=10: SlottedSphere
+//
+//  - Problem 1 solves the minimal surface equation of the form z=f(x,y),
+//              for the Dirichlet problem, using Picard iterations:
+//              -div( q grad u) = 0, with q(u) = (1 + |∇u|²)^{-1/2}
+//
+// Compile with: make minimal-surface
+//
+// Sample runs:  minimal-surface
+//               minimal-surface -a
+//               minimal-surface -c
+//               minimal-surface -c -a
+//               minimal-surface -no-pa
+//               minimal-surface -no-pa -a
+//               minimal-surface -no-pa -a -c
+//               minimal-surface -p 1
+//
+// Device sample runs:
+//               minimal-surface -d debug
+//               minimal-surface -d debug -a
+//               minimal-surface -d debug -c
+//               minimal-surface -d debug -c -a
+//               minimal-surface -d  cuda
+//               minimal-surface -d  cuda -a
+//               minimal-surface -d  cuda -c
+//               minimal-surface -d  cuda -c -a
+//               minimal-surface -d  cuda -no-pa
+//               minimal-surface -d  cuda -no-pa -a
+//               minimal-surface -d  cuda -no-pa -c
+//               minimal-surface -d  cuda -no-pa -c -a
+
+#include "mfem.hpp"
+#include "../../general/forall.hpp"
+
+using namespace mfem;
+
+// Constant variables
+constexpr int DIM = 2;
+constexpr int SDIM = 3;
+constexpr double PI = M_PI;
+constexpr double NRM = 1.e-4;
+constexpr double EPS = 1.e-14;
+constexpr Element::Type QUAD = Element::QUADRILATERAL;
+constexpr double NL_DMAX = std::numeric_limits<double>::max();
+
+// Static variables for GLVis
+static socketstream glvis;
+constexpr int GLVIZ_W = 1024;
+constexpr int GLVIZ_H = 1024;
+constexpr int  visport = 19916;
+constexpr char vishost[] = "localhost";
+
+// Context/Options for the solver
+struct Opt
+{
+   int sz, id;
+   int pb = 0;
+   int nx = 6;
+   int ny = 6;
+   int order = 3;
+   int refine = 2;
+   int niters = 8;
+   int surface = 5;
+   bool pa = true;
+   bool vis = true;
+   bool amr = false;
+   bool wait = false;
+   bool print = false;
+   bool radial = false;
+   bool by_vdim = false;
+   bool snapshot = false;
+   bool vis_mesh = false;
+   double tau = 1.0;
```

Comment:
Just out of curiosity, why is `tau` not available in parallel?

Reply:
I was using only in serial to create the gif, I pushed it to the parallel version too.